internal/controller/driver_controller.go

```diff
@@ -73,7 +73,7 @@ const ownerRefAnnotationKey = "csi.ceph.io/ownerref"
 
 // A regexp used to parse driver's prefix and type from the full name
 var nameRegExp, _ = regexp.Compile(fmt.Sprintf(
-	`^(?:(.+)\.)?(%s|%s|%s)\.csi\.ceph\.com$`,
+	`(%s|%s|%s)\.csi\.ceph\.com$`,
```

Comment:
This makes the regexp too permissive. It will allow names like 'mydriverrbd.csi.ceph.com`.
It is also preferable, for performance reasons, to compile a regexp that matches the entire name and not just the suffix. This way we eliminate a class of bugz that relates to greedy grepping 

```suggestion
	`^(?:.+\.)?(%s|%s|%s)\.csi\.ceph\.com$`,
```